src/Server/GRPCServer.cpp

```diff
@@ -484,19 +477,12 @@ namespace
                   grpc::ServerCompletionQueue & notification_queue,
                   const CompletionCallback & callback) override
         {
-            grpc_service.RequestExecuteQueryWithStreamOutput(&grpc_context, &query_info.emplace(), &writer, &new_call_queue, &notification_queue, getCallbackPtr(callback));
+            grpc_service.RequestExecuteQueryWithStreamOutput(&grpc_context, &query_info, &writer, &new_call_queue, &notification_queue, getCallbackPtr(callback));
         }
 
         void read(GRPCQueryInfo & query_info_, const CompletionCallback & callback) override
         {
-            if (!query_info.has_value())
-            {
-                callback(false);
-                return;
-            }
-
-            query_info_ = std::move(query_info).value();
-            query_info.reset();
+            query_info_ = std::move(query_info);
             callback(true);
```

Comment:
For several consecutive calls of `GRPCServer:: read` the `callback` will always be called with `true`, even though the `query_info` is a moved-from object (presumably reset by the move assignment operator).

Is this the indented behavior?